damask_parse/utils.py

```diff
@@ -534,6 +534,20 @@ def get_HDF5_incremental_quantity(hdf5_path, dat_path, transforms=None, incremen
         return all_data
 
 
+def normalise_inc(inc, incs_in_file, default=0):
+    """Convert a negative increment index into an increment integer."""
+    if inc < 0:
+        # Interpret as a negative index instead of an actual increment:
+        inc_idx = len(incs_in_file) + inc
+        try:
+            inc = incs_in_file[inc_idx]
```

Comment:
I'm not too sure about this. It will treat a negative inc based on only the increments in the file not all in the simulation. For me it would be better to find the final inc in the file and just take the -ve value from that so assuming the final inc in the file is the final inc of the file it counts back the actual sim increments. This is more consistent with the treatment of +ve values. Example, sim outputs on even increment 0,2,4,6,8 currently I think -3 would output from sim inc 4 of the sim where I think it would be better for it output from inc 6. What do you think @aplowman?

Reply:
I think you are right; better to be consistent if possible.

What negative values would you expect to correspond to each of the increments in your example (`incs_in_file = [0, 2, 4, 6, 8]`)? I can't immediately see how we would we choose the final increment via a negative number in this way(?)

The use-case here for me is when I just want to compare the initial and final increments, so I would want a way to choose the final increment without explicitly passing the final increment number. Thanks!

Reply:
To be consistent I would have it that -1 is the final inc in the file and then count backwards from there but by the sim inc number not position in the output file. Example if a sim output on incs `[0, 2, 4, 6, 8]` counting backwards these would be `[-9, -7, -5, -3, -1]`.

For your use case, you should be able to pass `values: [0, -1]` and get the first and final inc in the output file.

Does that sound reasonable to you? I will make the change if so.

Reply:
Yes, that sounds reasonable, thanks for clarifying! 👍